### ISSUE TITLE
Incorrect command example will cause change to tenant instead of sites

### ISSUE THREAD

**Issue by tylersoren** [MicrosoftDocs/OfficeDocs-SharePoint#3318]
I noticed an issue with the documented command for setting OneDrive sites to 'AllowLimitedAccess' as the conditional access policy. It is currently piping Get-SPOSite into Set-SPOTenant instead of Set-SPOSite.  With the way it is written it will set limited access for the entire tenant which is likely not what the administrator wanted to do and could cause lack of access to other Sharepoint resources outside of OneDrive.

this:
`Get-SPOSite -IncludePersonalSite $true -Limit all -Filter "Url -like '-my.sharepoint.com/personal/'" | Set-SPOTenant -ConditionalAccessPolicy AllowLimitedAccess`

should be:

`Get-SPOSite -IncludePersonalSite $true -Limit all -Filter "Url -like '-my.sharepoint.com/personal/'" | Set-SPOSite -ConditionalAccessPolicy AllowLimitedAccess`

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2f2abda5-9273-1a56-91b6-e4e77d5706b0
* Version Independent ID: 6a85f2a5-b892-55f7-b49a-4ab15f1e8b73
* Content: [SharePoint and OneDrive unmanaged device access controls for administrators - SharePoint in Microsoft 365](https://docs.microsoft.com/en-US/sharepoint/control-access-from-unmanaged-devices?WT.mc_id=365AdminCSH_spo#block-or-limit-access-to-a-specific-sharepoint-site-or-onedrive)
* Content Source: [SharePoint/SharePointOnline/control-access-from-unmanaged-devices.md](https://github.com/MicrosoftDocs/OfficeDocs-SharePoint/blob/public/SharePoint/SharePointOnline/control-access-from-unmanaged-devices.md)
* Service: **sharepoint-online**
* GitHub Login: @MikePlumleyMSFT
* Microsoft Alias: **mikeplum**

**Comment by get-itips** [comment]
Hello @tylersoren 

Please see here https://github.com/MicrosoftDocs/OfficeDocs-SharePoint/issues/2862

I'll test again if this has changed

**Comment by tylersoren** [comment]
After further investigation, it seems that it's not only the command in the documentation that is misleading, but the PowerShell cmdlets do not work as intended either. It appears true that using Set-SPOTenant doesn't make any changes to the tenant in certain circumstances, but at first glance, it doesn't look like it is updating the site policy either. Here is the output of running the command and then checking policy on the tenant and OneDrive sites.

![2022-06-13 08_14_52-Window](https://user-images.githubusercontent.com/33703139/173351697-5169d260-3181-4c4d-8099-fa9dc7da0e1a.png)

The issue you referenced[#2862]( https://github.com/MicrosoftDocs/OfficeDocs-SharePoint/issues/2862) mentioned that Set-SPOTenant did not accept pipeline input at the time. It looks like [accepting pipeline input was added shortly after](https://github.com/MicrosoftDocs/office-docs-powershell/pull/8207/files) but not for a site url, instead it's for the ConditionalAccessPolicy which should just be 'AllowFullAccess', 'AllowLimitedAccess' or 'BlockAccess'. 

While it doesn't make sense that this would have any effect at an individual site level, it actually does. Unfortunately, the Get-SPOSite command seems to be broken when trying to display the policy for multiple sites as a list. If instead you target an individual site with -Identity param, it will show that the site was updated. This can be achieved for all sites with a foreach loop as well.

Alternatively, if I use `Get-SPOSite -IncludePersonalSite $true -Limit all -Filter "Url -like '-my.sharepoint.com/personal/'" | Set-SPOSite -ConditionalAccessPolicy AllowLimitedAccess` and then list the sites with `Get-SPOSite -IncludePersonalSite $true`, it looks like it also does nothing. However, if I list the details of an individual site in the same way, then it shows the policy actually was updated.

![2022-06-13 08_47_39-Window](https://user-images.githubusercontent.com/33703139/173357263-e41c0650-c792-47dd-a0fe-6d706e87d13b.png)

It appears that both patterns work, but from an expected behavior perspective, it makes more sense that you would use Set-SPOSite than Set-SPOTenant since the intended behavior is only to update policy at a Site level. This falls in line with standard PowerShell practice for cmdlets with a 'Verb' only affecting the 'Noun' that it is targeting. Using Set-SPOTenant could potentially have unintended consequences of updating site and tenant level properties at the same time.

Also, the broken behavior of Get-SPOSite should be fixed or documented when it comes to listing policy for multiple sites.

**Comment by get-itips** [comment]
@tylersoren I'll check again to see if the behavior changed, in the meantime, please take into account that using the -Filter parameter outputs default values for some properties, one of them being ConditionalAccessPolicy, so you have to query per site to get the actual value (this is documented in [Get-SPOSite](https://docs.microsoft.com/en-us/powershell/module/sharepoint-online/get-sposite?view=sharepoint-ps) article)

I'll get back when I have the results

**Comment by tylersoren** [comment]
@get-itips, Thanks for looking into this.  I tried using Get-SPOSite without the filter or limit and get the same result so listing of any sort seems to be broken with the cmdlet if trying to get those extended properties.

![2022-06-13 09_03_18-Window](https://user-images.githubusercontent.com/33703139/173360292-8002abab-2f72-4a44-bbbe-81744b92781c.png)

 All show 'AllowFullAccess' even though they don't appear that way when querying individually.

**Comment by get-itips** [comment]
Hello @tylersoren how are you?

I tested again and piping into Set-SPOTenant still changes the ConditionalAccessPolicy of the Site (without changing the one from the Tenant).

And I also tested and it seems it nows also works piping into Set-SPOSite too.

It is also true that one can think it does nothing depending on the way the property is queried.

For clarity, I'll propose an update to the article.

Thanks for your feedback!
